core/state/database.go

```diff
@@ -132,23 +132,34 @@ func NewDatabase(db ethdb.Database) Database {
 func NewDatabaseWithConfig(db ethdb.Database, config *trie.Config) Database {
 	csc, _ := lru.New(codeSizeCacheSize)
 	return &cachingDB{
-		db:            trie.NewDatabaseWithConfig(db, config),
-		disk:          db,
+		db:            db,
+		triedb:        trie.NewDatabaseWithConfig(db, config),
+		codeSizeCache: csc,
+		codeCache:     fastcache.New(codeCacheSize),
+	}
+}
+
+// NewDatabaseWithNodeDB creates a state database with a live node database.
+func NewDatabaseWithNodeDB(db ethdb.Database, triedb *trie.Database) Database {
+	csc, _ := lru.New(codeSizeCacheSize)
+	return &cachingDB{
+		db:            db,
+		triedb:        triedb,
 		codeSizeCache: csc,
 		codeCache:     fastcache.New(codeCacheSize),
 	}
 }
 
 type cachingDB struct {
-	db            *trie.Database
-	disk          ethdb.KeyValueStore
+	db            ethdb.KeyValueStore
+	triedb        *trie.Database
```

Comment:
Hold on. So
`db` -> `triedb`
`disk` -> `db`

That seems a bit dangerous, to flat-out swap the names like that. I have a feeling that it could lead to mistakes. Perhaps let `disk` keep the name `disk` for now?